bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -450,31 +616,25 @@ impl From<ffi::evmc_result> for ExecutionResult {
     }
 }
 
-fn allocate_output_data(output: Option<&Vec<u8>>) -> (*const u8, usize) {
-    if let Some(buf) = output {
-        let buf_len = buf.len();
-
-        // Manually allocate heap memory for the new home of the output buffer.
-        let memlayout = std::alloc::Layout::from_size_align(buf_len, 1).expect("Bad layout");
-        let new_buf = unsafe { std::alloc::alloc(memlayout) };
-        unsafe {
-            // Copy the data into the allocated buffer.
-            std::ptr::copy(buf.as_ptr(), new_buf, buf_len);
+fn vec_into_boxed_slice_into_raw<T>(v: Option<Vec<T>>) -> (*const T, usize) {
+    match v {
+        Some(v) => {
+            let slice = Box::into_raw(v.into_boxed_slice());
+            (slice as *const T, slice.len())
         }
-
-        (new_buf as *const u8, buf_len)
-    } else {
-        (std::ptr::null(), 0)
+        None => (std::ptr::null(), 0),
     }
 }
 
-unsafe fn deallocate_output_data(ptr: *const u8, size: usize) {
-    if !ptr.is_null() {
-        let buf_layout = std::alloc::Layout::from_size_align(size, 1).expect("Bad layout");
-        std::alloc::dealloc(ptr as *mut u8, buf_layout);
+fn boxed_slice_from_raw_parts<T>(ptr: *const T, len: usize) -> Option<Box<[T]>> {
+    if ptr.is_null() {
+        None
+    } else {
+        Some(unsafe { Box::<[T]>::from_raw(slice::from_raw_parts_mut(ptr as *mut T, len)) })
     }
 }
```

Comment:
What about this? The old code works but IMO it is much cleaner to only convert between types and not invoke the allocator directly. Should this also be reverted?

Reply:
As we discussed offline, this code does not add any feature required for the stepN. And therefore this change may create a divergence between evmc and us. That could be problematic in the future. 
I think the code is good, but we should submit these changes, one at the time, to evmc itself. This is done so that when they do release a new version, we do not have to apply a huge patch to it when we want to catch up.